hpy/devel/src/runtime/ctx_type.c

```diff
@@ -574,11 +764,41 @@ create_slot_defs(HPyType_Spec *hpyspec, HPy_ssize_t base_member_offset,
             HPyDef *src = hpyspec->defines[i];
             if (src->kind != HPyDef_Kind_Slot || is_bf_slot(src))
                 continue;
-            if (is_destroy_slot(src)) {
+            if (is_slot(src, HPy_tp_destroy)) {
                 extra->tp_destroy_impl = (HPyFunc_destroyfunc)src->slot.impl;
                 continue;   /* we don't have a trampoline for tp_destroy */
             }
-            if (is_traverse_slot(src)) {
+            if (is_slot(src, HPy_tp_vectorcall_default)) {
+                /* Slot 'HPy_tp_vectorcall_default' will add a hidden field to
+                   the type's struct. The field can only be appended which
+                   conflicts with var objects. So, we don't allow this if
+                   itemsize != 0. */
+                if (hpyspec->itemsize) {
+                    PyMem_Free(result);
+                    PyErr_SetString(PyExc_TypeError,
+                            "Cannot use vectorcall protocol with var objects");
+                    return NULL;
```

Comment:
I think this problem might be solved by putting the hidden field *before* the user struct (i.e., at the end of the `PyObject_HEAD`), but I suppose that this would make the implementation of `HPy_AsStruct` more complex?

I am fine leaving this as is and fix it later when/if we need to implement vectorcall with varsized objects.